source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -0,0 +1,473 @@
+    }
+
+    void printSecondarySeparationLine()
+    {
+        std::cout << std::left << std::setfill('-') << std::setw(56) << "  " << std::endl;
+    }
+
+    void printCentered(const std::string &text)
+    {
+        constexpr size_t columns{ 56 };
+        std::cout << std::string((columns - text.size()) / 2, ' ') << text << std::endl;
+    }
+
+    void printTimeMedianMeanLine()
+    {
+        std::cout << std::left << std::setfill(' ') << std::setw(32) << "  Time:" << std::setw(13) << "Median"
+                  << "Mean" << std::endl;
+    }
+
+    void printResultLine(const std::string &name, Duration durationMedian, Duration durationMean)
+    {
+        std::cout << std::left << std::setfill(' ') << std::setw(32) << name << std::setw(13)
+                  << formatDuration(durationMedian) << formatDuration(durationMean) << std::endl;
+    }
+
+    void printConnectHeader(const size_t numConnects)
+    {
+        printPrimarySeparationLine();
+        std::cout << "Connecting and disconnecting " << numConnects << " times"
+                  << " (be patient):" << std::endl;
+    }
+
+    void printCapture3DHeader(const size_t numFrames, const std::vector<Zivid::Settings> &settingsVector)
+    {
+        const auto filterList = makefilterList(settingsVector);
+        printPrimarySeparationLine();
+        std::cout << "Capturing " << numFrames << " 3D frames:" << std::endl;
+        std::cout << "  exposure time = " << makeExposureTimeList(settingsVector) << std::endl;
+        std::cout << "  iris settings = " << makeIrisList(settingsVector) << std::endl;
+        if(!filterList.empty()) std::cout << "  filters = " << filterList << std::endl;
+        printSecondarySeparationLine();
+    }
+
+    void printCapture2DHeader(const size_t numFrames, const Zivid::Settings2D &settings)
+    {
+        printPrimarySeparationLine();
+        std::cout << "Capturing " << numFrames << " 2D frames:" << std::endl;
+        std::cout << "  exposure time = { " << settings.exposureTime().toString() << " }" << std::endl;
+    }
+
+    void printSaveHeader(const size_t numFrames)
+    {
+        printPrimarySeparationLine();
+        std::cout << "Saving point cloud " << numFrames << " times each"
+                  << " (be patient):" << std::endl;
+    }
+
+    void printResults(const std::vector<std::string> &names, const std::vector<std::vector<Duration>> &durations)
+    {
+        printSecondarySeparationLine();
+        printTimeMedianMeanLine();
+        for(size_t i = 0; i < durations.size(); i++)
+        {
+            printResultLine(names.at(i),
+                            computeMedianDuration(durations.at(i)),
+                            computeAverageDuration(durations.at(i)));
+        }
+    }
+
+    void printConnectResults(const std::vector<std::vector<Duration>> &durations)
+    {
+        const std::vector<std::string> names{ "  Connect:", "  Disconnect:" };
```

Comment:
Without an update below the results here are actually "time to connect" and "time to connect + disconnect"

Reply:
Good catch. The calculation code is wrong. Fixed now.